ical/parsing/parser.py

```diff
@@ -54,67 +39,127 @@
     PARSE_PARAMS,
     PARSE_VALUE,
 )
-from .unicode import SAFE_CHAR, VALUE_CHAR
-
 
 _LOGGER = logging.getLogger(__name__)
 
 
-@cache
-def _create_parser() -> ParserElement:
-    """Create rfc5545 parser."""
-    iana_token = Word(alphanums + "-")
-    vendor_id = Word(alphanums)
-    x_name = Combine("X-" + ZeroOrMore(vendor_id + "-") + alphas + nums + "-")
-    name = Or([iana_token, x_name])
-
-    param_name = Or([iana_token, x_name])
-    # rfc5545 Q-SAFE-CHAR is any character except CONTROL and DQUOTE which is
-    # close enough to the pyparsing provided parser element
-    param_text = Word(SAFE_CHAR) | ""
-    quoted_string = QuotedString('"')
-
-    param_value = Or([param_text, quoted_string])
-    # There are multiple levels of property parameter grouping since there can
-    # either be repreated property parameters with the same name or property
-    # s parameters with repeated values. A two level structure is used to grab
-    # both, which is then flattened when consuming the result.
-    param = Group(
-        param_name.set_results_name(PARSE_PARAM_NAME)
-        + "="
-        + param_value.set_results_name(PARSE_PARAM_VALUE, list_all_matches=True)
-        + ZeroOrMore(
-            "," + param_value.set_results_name(PARSE_PARAM_VALUE, list_all_matches=True)
-        )
-    ).set_results_name(PARSE_PARAMS, list_all_matches=True)
+_RE_CONTROL_CHARS = re.compile("[\x00-\x08\x0A-\x1F\x7F]")
+_RE_NAME = re.compile("[A-Z0-9\-]+")
 
-    contentline = (
-        name.set_results_name(PARSE_NAME)
-        + Group(ZeroOrMore(";" + param)).set_results_name(
-            PARSE_PARAMS, list_all_matches=True
+@cache
```

Comment:
Curious what kind of performance benefits we see vs memory usage of caching this. Maybe an LRU with fixed size could be more appropriate here.

Reply:
I'll re-add the caching decorator

Reply:
I'm worried about the memory usage of doing this. Whats the performance with and without? What does it look like with an LRU?